### PR TITLE
Make EvolveNewtonianEuler handle analytic data

### PR DESCRIPTION
**Pull request by carmaza**
- Template Metavariables in initial data type
- Allow evolution of analytic data by making analytic BCs and ObserveErrorNorms optional
- Executable is now compiled by `make`ing Evolve + name of the initial data class

These changes will make evolving analytic data straightforward, in the same way `ValenciaDivClean` does.

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Evolution/Executables/NewtonianEuler/EvolveNewtonianEuler.hpp [hunk 1]
```diff
@@ -82,10 +83,10 @@ class CProxy_ConstGlobalCache;
 }  // namespace Parallel
 /// \endcond
 
-template <size_t Dim>
+template <size_t Dim, typename InitialData>
 struct EvolutionMetavars {
   static constexpr size_t volume_dim = Dim;
-  using initial_data = NewtonianEuler::Solutions::RiemannProblem<Dim>;
+  using initial_data = InitialData;
```
Comment (nilsdeppe) [inline src/Evolution/Executables/NewtonianEuler/EvolveNewtonianEuler.hpp:94]: Please add the `static_assert` that's also in the GRMHD executable.

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
I'm fine with you squashing immediately

**Comment by carmaza** [comment]
Done!

**Review by nilsdeppe — APPROVED** [review]
LGTM! :+1:

**Review by kidder — APPROVED** [review]

**Pull request merged**